pkg/reconciler/build/build.go

```diff
@@ -158,10 +158,23 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 		return c.cancelBuild(build, logger)
 	}
 
+	// If build creation already started, then ignore it.
+	if isStarted(&build.Status) {
+		return nil
+	}
+
 	// If the build hasn't started yet, validate it and create a Pod for it
 	// and record that pod's name in the build status.
 	var p *corev1.Pod
-	if build.Status.Cluster == nil || build.Status.Cluster.PodName == "" {
+	if build.Status.StartTime == nil {
+		build.Status.SetCondition(&duckv1alpha1.Condition{
```

Comment:
Would it be sufficient to set the `StartTime` here and update the status immediately? AIUI we basically just need to ensure this branch isn't taken for the same build twice.

I think we could also accomplish that by generating a `podName` and persisting it immediately (then checking if `podName != ""`), but users would sometimes see a pod name that doesn't exist yet, which could be confusing.

Reply:
> which could be confusing.

True but the build is still in "Unknown" state so it doesn't seem that bad.

Reply:
> Would it be sufficient to set the `StartTime` here and update the status immediately?
that may work as well. If we agree that dummy pod name should be removed, I can try this

> True but the build is still in "Unknown" state so it doesn't seem that bad.
I think it makes build statuses very unreliable - `Unknown` means that we can't trust build pod name that we get from API and `succeeded` means that build already passed. What if we need to "watch" build pod while build is not succeeded?